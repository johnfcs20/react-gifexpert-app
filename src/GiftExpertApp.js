import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GiftExpertApp = ({defaultCategory = []}) => {

     const [categories, setCategories] =  useState (defaultCategory);
     
    return (
        <>
            <h1>GiftExpertApp</h1>
            <AddCategory setCategories={setCategories}/>
            <hr />
            <ol>
                {
                    categories.map(category =>
                     <GifGrid key={category} category={category} />
                    )
                }
            </ol>

        </>
    )
}