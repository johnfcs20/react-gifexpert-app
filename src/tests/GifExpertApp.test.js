import React from "react"
import { shallow } from "enzyme"
import { GiftExpertApp } from "../GiftExpertApp"

describe('Pruebas en <GiftExpertApp />', () => { 
    
    test('Debe mostrarse correctamente', () => { 
        const wrapper = shallow(<GiftExpertApp />);
        expect(wrapper).toMatchSnapshot();
     })

     test('Debe mostrar una lista de categorias', () => { 
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GiftExpertApp defaultCategory={categories} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

     })

 })