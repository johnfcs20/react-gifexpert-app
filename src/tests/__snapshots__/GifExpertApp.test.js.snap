// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftExpertApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GiftExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GiftExpertApp /> Debe mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    GiftExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
